power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -33,13 +42,63 @@ inline auto build_transformer_graph(State const& /*state*/) -> TransformerGraph
     return {};
 }
 
-inline auto get_edge_weights(TransformerGraph const& /*graph*/) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
-    // TODO(mgovers): implement
-    return {};
+inline auto get_edge_weights(TransformerGraph const& graph) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
+    // Step 2: Initialize the rank of all vertices (transformer nodes) as infinite (INT_MAX)
+    std::vector<EdgeWeight> rank(boost::num_vertices(graph), infty);
+    std::vector<Idx2D> sources(boost::num_vertices(graph));
+
+    // Step 3: Loop all the connected sources (status == 1)
+    for (auto v : boost::make_iterator_range(boost::vertices(graph))) {
+        if (graph[v].status == 1) {
+            // a. Perform Dijkstra shortest path algorithm from the vertex with that source.
+            // This is to determine the shortest path of all vertices to this particular source.
+            std::priority_queue<std::pair<EdgeWeight, TrafoGraphIdx>, std::vector<std::pair<EdgeWeight, TrafoGraphIdx>>,
+                                std::greater<>>
+                pq;
```

Comment:
maybe create a `using` declaration for this to make it more readable (may be within this function as it's very specific)

```suggestion
            using TrafoGraphElement = std::pair<EdgeWeight, TrafoGraphIdx>;
            std::priority_queue<TrafoGraphElement, std::vector<TrafoGraphElement>, std::greater<>> pq;
```